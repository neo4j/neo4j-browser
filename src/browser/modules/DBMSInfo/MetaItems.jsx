/*
 * Copyright (c) 2002-2019 "Neo4j,"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react'
import { ecsapeCypherMetaItem } from 'services/utils'
import classNames from 'classnames'
import styles from './style_meta.css'
import {
  DrawerSubHeader,
  DrawerSection,
  DrawerSectionBody
} from 'browser-components/drawer'
import {
  StyledLabel,
  StyledRelationship,
  StyledProperty,
  StyledShowMoreContainer,
  StyledShowMoreLink
} from './styled'
import Render from 'browser-components/Render'
import numberToUSLocale from 'shared/utils/number-to-US-locale'

const ShowMore = ({ total, shown, moreStep, onMore }) => {
  const numMore = total - shown > moreStep ? moreStep : total - shown
  return (
    <Render if={shown < total}>
      <StyledShowMoreContainer>
        <StyledShowMoreLink onClick={() => onMore(numMore)}>
          Show {numMore} more
        </StyledShowMoreLink>
        &nbsp;|&nbsp;
        <StyledShowMoreLink onClick={() => onMore(total)}>
          Show all
        </StyledShowMoreLink>
      </StyledShowMoreContainer>
    </Render>
  )
}

const createItems = (
  originalList,
  onItemClick,
  RenderType,
  editorCommandTemplate,
  showStar = true,
  count
) => {
  let items = [...originalList]
  if (showStar) {
    let str = '*'
    if (count) {
      str = `${str}(${numberToUSLocale(count)})`
    }
    items.unshift(str)
  }
  return items.map((text, index) => {
    const getNodesCypher = editorCommandTemplate(text, index)
    return (
      <RenderType.component
        data-testid='sidebarMetaItem'
        key={index}
        onClick={() => onItemClick(getNodesCypher)}
      >
        {text}
      </RenderType.component>
    )
  })
}
const LabelItems = ({
  labels = [],
  totalNumItems,
  onItemClick,
  moreStep,
  onMoreClick,
  count
}) => {
  let labelItems = <p>There are no labels in database</p>
  if (labels.length) {
    const editorCommandTemplate = (text, i) => {
      if (i === 0) {
        return 'MATCH (n) RETURN n LIMIT 25'
      }
      return `MATCH (n:${ecsapeCypherMetaItem(text)}) RETURN n LIMIT 25`
    }
    labelItems = createItems(
      labels,
      onItemClick,
      { component: StyledLabel },
      editorCommandTemplate,
      true,
      count
    )
  }
  return (
    <DrawerSection>
      <DrawerSubHeader>Node Labels</DrawerSubHeader>
      <DrawerSectionBody
        className={classNames({
          [styles['wrapper']]: true
        })}
      >
        {labelItems}
      </DrawerSectionBody>
      <ShowMore
        total={totalNumItems}
        shown={labels.length}
        moreStep={moreStep}
        onMore={onMoreClick}
      />
    </DrawerSection>
  )
}
const RelationshipItems = ({
  relationshipTypes = [],
  totalNumItems,
  onItemClick,
  moreStep,
  onMoreClick,
  count
}) => {
  let relationshipItems = <p>No relationships in database</p>
  if (relationshipTypes.length > 0) {
    const editorCommandTemplate = (text, i) => {
      if (i === 0) {
        return 'MATCH p=()-->() RETURN p LIMIT 25'
      }
      return `MATCH p=()-[r:${ecsapeCypherMetaItem(
        text
      )}]->() RETURN p LIMIT 25`
    }
    relationshipItems = createItems(
      relationshipTypes,
      onItemClick,
      { component: StyledRelationship },
      editorCommandTemplate,
      true,
      count
    )
  }
  return (
    <DrawerSection>
      <DrawerSubHeader>Relationship Types</DrawerSubHeader>
      <DrawerSectionBody
        className={classNames({
          [styles['wrapper']]: true
        })}
      >
        {relationshipItems}
      </DrawerSectionBody>
      <ShowMore
        total={totalNumItems}
        shown={relationshipTypes.length}
        moreStep={moreStep}
        onMore={onMoreClick}
      />
    </DrawerSection>
  )
}
const PropertyItems = ({
  properties,
  totalNumItems,
  onItemClick,
  moreStep,
  onMoreClick
}) => {
  let propertyItems = <p>There are no properties in database</p>
  if (properties.length > 0) {
    const editorCommandTemplate = text => {
      return `MATCH (n) WHERE EXISTS(n.${ecsapeCypherMetaItem(
        text
      )}) RETURN DISTINCT "node" as entity, n.${ecsapeCypherMetaItem(
        text
      )} AS ${ecsapeCypherMetaItem(
        text
      )} LIMIT 25 UNION ALL MATCH ()-[r]-() WHERE EXISTS(r.${ecsapeCypherMetaItem(
        text
      )}) RETURN DISTINCT "relationship" AS entity, r.${ecsapeCypherMetaItem(
        text
      )} AS ${ecsapeCypherMetaItem(text)} LIMIT 25`
    }
    propertyItems = createItems(
      properties,
      onItemClick,
      { component: StyledProperty },
      editorCommandTemplate,
      false
    )
  }
  return (
    <DrawerSection>
      <DrawerSubHeader>Property Keys</DrawerSubHeader>
      <DrawerSectionBody
        className={classNames({
          [styles['wrapper']]: true
        })}
      >
        {propertyItems}
      </DrawerSectionBody>
      <ShowMore
        total={totalNumItems}
        shown={properties.length}
        moreStep={moreStep}
        onMore={onMoreClick}
      />
    </DrawerSection>
  )
}

export { LabelItems, RelationshipItems, PropertyItems }
