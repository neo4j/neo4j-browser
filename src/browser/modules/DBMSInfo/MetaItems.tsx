/*
 * Copyright (c) "Neo4j"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
import deepmerge from 'deepmerge'
import React from 'react'
import { ThemeProvider } from 'styled-components'

import { GraphStyleModel } from 'graph-visualization'

import styles from './style_meta.css'
import { StyledLabel, StyledProperty, StyledRelationship } from './styled'
import { ShowMoreOrAll } from 'browser-components/ShowMoreOrAll/ShowMoreOrAll'
import {
  DrawerSection,
  DrawerSectionBody,
  DrawerSubHeader
} from 'browser-components/drawer/drawer-styled'
import { dark } from 'browser-styles/themes'
import { escapeCypherIdentifier } from 'services/utils'
import numberToUSLocale from 'shared/utils/number-to-US-locale'

const wrapperStyle = (styles && styles.wrapper) || ''

function createStyleGetter(graphStyleData: any, kind: string) {
  const graphStyle = new GraphStyleModel()
  if (graphStyleData) {
    graphStyle.loadRules(deepmerge(graphStyle.toSheet(), graphStyleData || {}))
  }

  if (kind === 'node') {
    return function (text: string) {
      if (graphStyleData) {
        const styleForItem = graphStyle.forNode({
          labels: [text]
        })
        return {
          backgroundColor: styleForItem.get('color'),
          color: styleForItem.get('text-color-internal')
        }
      }
      return {}
    }
  }
  if (kind === 'relationship') {
    return function (text: string) {
      if (graphStyleData) {
        const styleForItem = graphStyle.forRelationship({
          type: text
        })
        return {
          backgroundColor: styleForItem.get('color'),
          color: styleForItem.get('text-color-internal')
        }
      }
      return {}
    }
  }
  throw new Error(`Unsupported argument: ${kind}`)
}
function createNodeStyleGetter(graphStyleData: any) {
  return createStyleGetter(graphStyleData, 'node')
}
function createRelationshipStyleGetter(graphStyleData: any) {
  return createStyleGetter(graphStyleData, 'relationship')
}

const createItems = (
  originalList: any,
  onItemClick: any,
  RenderType: any,
  editorCommandTemplate: any,
  showStar = true,
  count?: number,
  styleGetter: any = () => ({})
) => {
  const items = [...originalList]
  if (showStar) {
    let str = '*'
    if (count) {
      str = `${str}(${numberToUSLocale(count)})`
    }
    items.unshift(str)
  }

  return items.map((text, index) => {
    const getNodesCypher = editorCommandTemplate(text, index)
    return (
      <RenderType.component
        data-testid="sidebarMetaItem"
        key={index}
        onClick={() => onItemClick(getNodesCypher)}
        style={styleGetter(text)}
      >
        {text}
      </RenderType.component>
    )
  })
}

type LabelItemsProps = {
  labels: string[]
  totalNumItems: number
  onItemClick: () => void
  moreStep: number
  onMoreClick: (num: number) => void
  count: number
  graphStyleData: any
}
const LabelItems = ({
  labels = [],
  totalNumItems,
  onItemClick,
  moreStep,
  onMoreClick,
  count,
  graphStyleData
}: LabelItemsProps) => {
  let labelItems: any = <p>There are no labels in database</p>
  if (labels.length) {
    const editorCommandTemplate = (text: any, i: any) => {
      if (i === 0) {
        return 'MATCH (n) RETURN n LIMIT 25'
      }
      return `MATCH (n:${escapeCypherIdentifier(text)}) RETURN n LIMIT 25`
    }
    labelItems = createItems(
      labels,
      onItemClick,
      { component: StyledLabel },
      editorCommandTemplate,
      true,
      count,
      createNodeStyleGetter(graphStyleData)
    )
  }
  return (
    <DrawerSection>
      <DrawerSubHeader>Node Labels</DrawerSubHeader>
      <DrawerSectionBody className={wrapperStyle}>
        {labelItems}
      </DrawerSectionBody>
      <ThemeProvider theme={dark}>
        <ShowMoreOrAll
          total={totalNumItems}
          shown={labels.length}
          moreStep={moreStep}
          onMore={onMoreClick}
        />
      </ThemeProvider>
    </DrawerSection>
  )
}

type RelationshipItemsProps = {
  relationshipTypes: string[]
  totalNumItems: number
  onItemClick: () => void
  moreStep: number
  onMoreClick: (num: number) => any
  count: number
  graphStyleData: any
}
const RelationshipItems = ({
  relationshipTypes = [],
  totalNumItems,
  onItemClick,
  moreStep,
  onMoreClick,
  count,
  graphStyleData
}: RelationshipItemsProps) => {
  let relationshipItems: any = <p>No relationships in database</p>
  if (relationshipTypes.length > 0) {
    const editorCommandTemplate = (text: any, i: any) => {
      if (i === 0) {
        return 'MATCH p=()-->() RETURN p LIMIT 25'
      }
      return `MATCH p=()-[r:${escapeCypherIdentifier(
        text
      )}]->() RETURN p LIMIT 25`
    }
    relationshipItems = createItems(
      relationshipTypes,
      onItemClick,
      { component: StyledRelationship },
      editorCommandTemplate,
      true,
      count,
      createRelationshipStyleGetter(graphStyleData)
    )
  }
  return (
    <DrawerSection>
      <DrawerSubHeader>Relationship Types</DrawerSubHeader>
      <DrawerSectionBody className={wrapperStyle}>
        {relationshipItems}
      </DrawerSectionBody>
      <ShowMoreOrAll
        total={totalNumItems}
        shown={relationshipTypes.length}
        moreStep={moreStep}
        onMore={onMoreClick}
      />
    </DrawerSection>
  )
}

type PropertyItemsProps = {
  properties: string[]
  totalNumItems: number
  onItemClick: () => void
  moreStep: number
  onMoreClick: (num: number) => any
}
const PropertyItems = ({
  properties,
  totalNumItems,
  onItemClick,
  moreStep,
  onMoreClick
}: PropertyItemsProps) => {
  let propertyItems: any = <p>There are no properties in database</p>
  if (properties.length > 0) {
    const editorCommandTemplate = (text: any) => {
      return `MATCH (n) WHERE EXISTS(n.${escapeCypherIdentifier(
        text
      )}) RETURN DISTINCT "node" as entity, n.${escapeCypherIdentifier(
        text
      )} AS ${escapeCypherIdentifier(
        text
      )} LIMIT 25 UNION ALL MATCH ()-[r]-() WHERE EXISTS(r.${escapeCypherIdentifier(
        text
      )}) RETURN DISTINCT "relationship" AS entity, r.${escapeCypherIdentifier(
        text
      )} AS ${escapeCypherIdentifier(text)} LIMIT 25`
    }
    propertyItems = createItems(
      properties,
      onItemClick,
      { component: StyledProperty },
      editorCommandTemplate,
      false
    )
  }
  return (
    <DrawerSection>
      <DrawerSubHeader>Property Keys</DrawerSubHeader>
      <DrawerSectionBody className={wrapperStyle}>
        {propertyItems}
      </DrawerSectionBody>
      <ShowMoreOrAll
        total={totalNumItems}
        shown={properties.length}
        moreStep={moreStep}
        onMore={onMoreClick}
      />
    </DrawerSection>
  )
}

export { LabelItems, RelationshipItems, PropertyItems }
