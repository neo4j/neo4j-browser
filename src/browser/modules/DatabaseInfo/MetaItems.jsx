/*
 * Copyright (c) 2002-2017 "Neo Technology,"
 * Network Engine for Objects in Lund AB [http://neotechnology.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import classNames from 'classnames'
import styles from './style_meta.css'
import {DrawerSubHeader, DrawerSection, DrawerSectionBody} from 'browser-components/drawer'
import {StyledLabel, StyledRelationship, StyledProperty} from './styled'

const createItems = (originalList, onItemClick, RenderType, editorCommandTemplate, showStar = true) => {
  let items = [...originalList]
  if (showStar) {
    items.unshift('*')
  }
  return items.map((text, index) => {
    const getNodesCypher = editorCommandTemplate(text)
    return (
      <RenderType.component
        key={index}
        onClick={() => onItemClick(getNodesCypher)}>
        {text}
      </RenderType.component>
    )
  })
}
const LabelItems = ({labels, onItemClick}) => {
  let labelItems = <p>There are no labels in database</p>
  if (labels.length > 0) {
    const editorCommandTemplate = (text) => {
      if (text === '*') {
        return 'MATCH (n) RETURN n LIMIT 25'
      }
      return `MATCH (n:${text}) RETURN n LIMIT 25`
    }
    labelItems = createItems(labels, onItemClick, {component: StyledLabel}, editorCommandTemplate)
  }
  return (
    <DrawerSection>
      <DrawerSubHeader>Node Labels</DrawerSubHeader>
      <DrawerSectionBody className={classNames({
        [styles['wrapper']]: true
      })}>
        {labelItems}
      </DrawerSectionBody>
    </DrawerSection>
  )
}
const RelationshipItems = ({relationshipTypes, onItemClick}) => {
  let relationshipItems = <p>No relationships in database</p>
  if (relationshipTypes.length > 0) {
    const editorCommandTemplate = (text) => {
      if (text === '*') {
        return 'MATCH p=()-->() RETURN p LIMIT 25'
      }
      return `MATCH p=()-[r:${text}]->() RETURN p LIMIT 25`
    }
    relationshipItems = createItems(relationshipTypes, onItemClick, {component: StyledRelationship}, editorCommandTemplate)
  }
  return (
    <DrawerSection>
      <DrawerSubHeader>Relationship Types</DrawerSubHeader>
      <DrawerSectionBody className={classNames({
        [styles['wrapper']]: true
      })}>
        {relationshipItems}
      </DrawerSectionBody>
    </DrawerSection>
  )
}
const PropertyItems = ({properties, onItemClick}) => {
  let propertyItems = <p>There are no properites in database</p>
  if (properties.length > 0) {
    const editorCommandTemplate = (text) => {
      return `MATCH (n) WHERE EXISTS(n.${text}) RETURN DISTINCT "node" as entity, n.${text} AS ${text} LIMIT 25 UNION ALL MATCH ()-[r]-() WHERE EXISTS(r.${text}) RETURN DISTINCT "relationship" AS entity, r.${text} AS ${text} LIMIT 25`
    }
    propertyItems = createItems(properties, onItemClick, {component: StyledProperty}, editorCommandTemplate, false)
  }
  return (
    <DrawerSection>
      <DrawerSubHeader>Property Keys</DrawerSubHeader>
      <DrawerSectionBody className={classNames({
        [styles['wrapper']]: true
      })}>
        {propertyItems}
      </DrawerSectionBody>
    </DrawerSection>
  )
}

export {
  LabelItems,
  RelationshipItems,
  PropertyItems
}
