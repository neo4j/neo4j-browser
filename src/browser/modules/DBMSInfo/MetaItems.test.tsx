/*
 * Copyright (c) "Neo4j"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
import { render, screen } from '@testing-library/react'
import React from 'react'

import { LabelItems, PropertyItems, RelationshipItems } from './MetaItems'

const renderLabelItems = (
  items: string[] = [],
  moreStep = 5,
  totalNumItems: number = items.length
) => {
  return render(
    <LabelItems
      labels={items}
      count={5}
      graphStyleData={''}
      moreStep={moreStep}
      onItemClick={jest.fn()}
      onMoreClick={jest.fn()}
      totalNumItems={totalNumItems}
    />
  )
}

const renderPropertyItems = (
  items: string[] = [],
  moreStep = 5,
  totalNumItems: number = items.length
) => {
  return render(
    <PropertyItems
      properties={items}
      moreStep={moreStep}
      onItemClick={jest.fn()}
      onMoreClick={jest.fn()}
      totalNumItems={totalNumItems}
    />
  )
}

const renderRelationshipItems = (
  items: string[] = [],
  moreStep = 5,
  totalNumItems: number = items.length
) => {
  return render(
    <RelationshipItems
      relationshipTypes={items}
      moreStep={moreStep}
      onItemClick={jest.fn()}
      onMoreClick={jest.fn()}
      totalNumItems={totalNumItems}
      count={5}
      graphStyleData={''}
    />
  )
}

test('LabelItems renders empty', () => {
  const { container } = renderLabelItems([])
  expect(container).toMatchSnapshot()
})
test('LabelItems renders labels', () => {
  const items = ['MyLabel', 'MyLabel2']
  const { container } = renderLabelItems(items)
  expect(container).toMatchSnapshot()
})
test('RelationshipItems renders empty', () => {
  const { container } = renderRelationshipItems([])
  expect(container).toMatchSnapshot()
})
test('RelationshipItems renders relationshipTypes', () => {
  const items = ['MY_TYPE', 'MY_TYPE2']
  const { container } = renderRelationshipItems(items)
  expect(container).toMatchSnapshot()
})
test('PropertyItems renders empty', () => {
  const { container } = renderPropertyItems([])
  expect(container).toMatchSnapshot()
})
test('PropertyItems renders properties', () => {
  const items = ['prop1', 'prop2']
  const { container } = renderPropertyItems(items)
  expect(container).toMatchSnapshot()
})
test('PropertyItems with more than step size properties shows "Show all" link', () => {
  const items = Array.from({ length: 6 }).map(
    (_value: any, index: number) => `prop${index}`
  )
  renderPropertyItems(items, 5, 10)
  expect(screen.getByText('Show all')).toBeInTheDocument()
})
