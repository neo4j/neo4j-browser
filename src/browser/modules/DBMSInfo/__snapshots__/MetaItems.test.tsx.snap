// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelItems renders empty 1`] = `
<div>
  <div
    class="sc-iAKWXU fiffMu"
  >
    <h5
      class="sc-iUKqMP kWpjPs"
    >
      Node labels
    </h5>
    <div
      class="sc-efQSVx ka-Dmgl"
    >
      <p>
        There are no labels in database
      </p>
    </div>
  </div>
</div>
`;

exports[`LabelItems renders labels 1`] = `
<div>
  <div
    class="sc-iAKWXU fiffMu"
  >
    <h5
      class="sc-iUKqMP kWpjPs"
    >
      Node labels
    </h5>
    <div
      class="sc-efQSVx ka-Dmgl"
    >
      <div
        class="sc-dkPtRN sc-hKwDye jYhbHV gKvrSp"
        data-testid="sidebarMetaItem"
      >
        *(5)
      </div>
      <div
        class="sc-dkPtRN sc-hKwDye jYhbHV gKvrSp"
        data-testid="sidebarMetaItem"
      >
        MyLabel
      </div>
      <div
        class="sc-dkPtRN sc-hKwDye jYhbHV gKvrSp"
        data-testid="sidebarMetaItem"
      >
        MyLabel2
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertyItems renders empty 1`] = `
<div>
  <div
    class="sc-iAKWXU fiffMu"
  >
    <h5
      class="sc-iUKqMP kWpjPs"
    >
      Property keys
    </h5>
    <div
      class="sc-efQSVx ka-Dmgl"
    >
      <p>
        There are no properties in database
      </p>
    </div>
  </div>
</div>
`;

exports[`PropertyItems renders properties 1`] = `
<div>
  <div
    class="sc-iAKWXU fiffMu"
  >
    <h5
      class="sc-iUKqMP kWpjPs"
    >
      Property keys
    </h5>
    <div
      class="sc-efQSVx ka-Dmgl"
    >
      <div
        class="sc-dkPtRN sc-jRQBWg jYhbHV jZJEwZ"
        data-testid="sidebarMetaItem"
      >
        prop1
      </div>
      <div
        class="sc-dkPtRN sc-jRQBWg jYhbHV jZJEwZ"
        data-testid="sidebarMetaItem"
      >
        prop2
      </div>
    </div>
  </div>
</div>
`;

exports[`RelationshipItems renders empty 1`] = `
<div>
  <div
    class="sc-iAKWXU fiffMu"
  >
    <h5
      class="sc-iUKqMP kWpjPs"
    >
      Relationship types
    </h5>
    <div
      class="sc-efQSVx ka-Dmgl"
    >
      <p>
        No relationships in database
      </p>
    </div>
  </div>
</div>
`;

exports[`RelationshipItems renders relationshipTypes 1`] = `
<div>
  <div
    class="sc-iAKWXU fiffMu"
  >
    <h5
      class="sc-iUKqMP kWpjPs"
    >
      Relationship types
    </h5>
    <div
      class="sc-efQSVx ka-Dmgl"
    >
      <div
        class="sc-dkPtRN sc-eCImPb jYhbHV dPdKNs"
        data-testid="sidebarMetaItem"
      >
        *(5)
      </div>
      <div
        class="sc-dkPtRN sc-eCImPb jYhbHV dPdKNs"
        data-testid="sidebarMetaItem"
      >
        MY_TYPE
      </div>
      <div
        class="sc-dkPtRN sc-eCImPb jYhbHV dPdKNs"
        data-testid="sidebarMetaItem"
      >
        MY_TYPE2
      </div>
    </div>
  </div>
</div>
`;
